PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheck.java

```diff
@@ -0,0 +1,241 @@
+
+        // Define constants for string literals
+        private static final RuleConfig ruleConfig;
+        private static final String SUGGESTION;
+        private static final List<String> METHODS_TO_CHECK;
+        private static final List<String> CLIENTS_TO_CHECK;
+        private static boolean SKIP_WHOLE_RULE;
+
+        private static final Logger LOGGER = Logger.getLogger(ServiceBusReceiveModeCheck.class.getName());
+
+        static {
+            final String ruleName = "ServiceBusReceiveModeCheck";
+            RuleConfigLoader centralRuleConfigLoader = RuleConfigLoader.getInstance();
+
+            // Get the RuleConfig object for the rule
+            ruleConfig = centralRuleConfigLoader.getRuleConfig(ruleName);
+
+            METHODS_TO_CHECK = ruleConfig.getMethodsToCheck();
+            CLIENTS_TO_CHECK = ruleConfig.getClientsToCheck();
+            SUGGESTION = ruleConfig.getAntiPatternMessage();
+            SKIP_WHOLE_RULE = ruleConfig == RuleConfig.EMPTY_RULE || METHODS_TO_CHECK.isEmpty();
+        }
+
+        /**
+         * Constructor for the visitor
+         *
+         * @param holder The holder for the problems found
+         */
+        public ServiceBusReceiveModeVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+        }
+
+        /**
+         * This method is used to visit the declaration statement in the code.
+         * It checks for the declaration of the Azure SDK ServiceBusReceiver & ServiceBusProcessor clients
+         * and whether the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+         * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1,
+         * a problem is registered with the ProblemsHolder.
+         *
+         * @param statement The declaration statement to visit
+         */
+        @Override
+        public void visitDeclarationStatement(PsiDeclarationStatement statement) {
+            super.visitDeclarationStatement(statement);
+
+            if (SKIP_WHOLE_RULE) {
+                return;
+            }
+
+            // Get the declared elements
+            PsiElement[] elements = statement.getDeclaredElements();
+
+            // Get the variable declaration
+            if (elements.length > 0 && elements[0] instanceof PsiVariable) {
+                PsiVariable variable = (PsiVariable) elements[0];
+
+                // Process the variable declaration
+                processVariableDeclaration(variable);
+            }
+        }
+
+        /**
+         * This method is used to process the variable declaration.
+         * It checks if the client type is an Azure ServiceBus client and
+         * retrieves the client name (left side of the declaration).
+         *
+         * @param variable The variable to process
+         */
+        private void processVariableDeclaration(PsiVariable variable) {
```

Comment:
This method is called once. You want to create methods as pieces of code you see reused in the same class.  I would roll this back up into the `visitDeclearationStatement` method above.

Reply:
I made it a separate method because Sameeksha mentioned moving away from overly chunky code in one of the first PRs -- so I've been doing separate methods when the flow becomes long

Reply:
Personal preference I guess. ;P